src/js/tabs/firmware_flasher.js

```diff
@@ -264,44 +276,57 @@ TABS.firmware_flasher.initialize = function (callback) {
             let releases = {};
             let unifiedConfigs = {};
             let items = {};
-            let baseTargets =  {};
+            let unifiedTargetNames = [];
             data.forEach(function(target) {
-                let targetName = target.name;
-                if (targetName.endsWith('.config')) {
-                    targetName = targetName.slice(0,targetName.indexOf('.config'));
-                } else {
+                const TARGET_REGEXP = /^(?:([^-]{1,4})-)?(.*).config$/;
+                let targetParts = target.name.match(TARGET_REGEXP);
+                if (!targetParts) {
                     return;
                 }
-                unifiedConfigs[targetName]=target.download_url;
-                items[targetName] = "something";
+                let boardName = targetParts[2];
+                let manufacturerId = targetParts[1];
+                let targetName;
+                let displayName;
+                if (manufacturerId) {
+                    targetName = `${boardName}+${manufacturerId}`;
+                    displayName = `${boardName} (${manufacturerId})`;
+                } else {
+                    targetName = boardName;
+                }
+                unifiedTargetNames.push(boardName);
+                unifiedConfigs[targetName] = target.download_url;
+                items[targetName] = { displayName: displayName };
+                // Chicken and egg problem: We need to know what Unified Target this configuration uses before reading the configuration.
+                // Solving this by assuming that all Unified Targets have the same availability for now.
+                const DEFAULT_UNIFIED_TARGET_NAME = "STM32F405";
+                releases[targetName] = builds[DEFAULT_UNIFIED_TARGET_NAME];
```

Comment:
Is that needed? for a unified target I thought it was 
1) grab the file, parse the UNIFIED_TARGET_NAME
2) Populate versions with those releases, eg releases[STM32F411]. Otherwise the unified only targets would have shown issues.

https://github.com/betaflight/betaflight-configurator/blob/074bb46f788e15a552a2267f433e3f21707c7f64/src/js/tabs/firmware_flasher.js#L550-L558

Reply:
Something that yields the available versions is needed.
The current way of using the non-unified .hex for this is not viable, as one goal of this whole exercise is to stop building / releasing them as soon as the Unified Target configurations are confirmed working for users, so the available versions for the non-unified targets will stop for some 4.1.x.
To get the actually used Unified Target .hex from the Unified Target config requires loading the file. If done when generating the list this **will** trigger GitHub's API request limiting (and be inefficient). If done when selecting a Unified Target configuration will introduce a delay **and** still trigger the request limit if a user is 'browsing'. So this is a pragmatic solution that will work for now.

Reply:
Grabbing the individual config files isn't doing API calls. So I didn't worry about how many requests I do.

We're doing an API call to get the contents of `configs/default` (list of files) and another one for the releases. In the API call for releases, we get the last 30 releases, which is chock full of data.

If you did a release with only unified targets, users would get the proper versions, its just that the (Legacy) option wouldn't include a latest version.

Reply:
> We're doing an API call to get the contents of configs/default (list of files) and another one for the releases. In the API call for releases, we get the last 30 releases, which is chock full of data.

But we get them on a per target basis. And if we use a target that has been replaced by a Unified Target configuration and isn't released any more to populate the Unified Target configuration's releases, there won't be any releases showing for after we've pulled the non-unified target from release.

Reply:
If you download the releases, and remove the 4.1.0 RC builds for one target it looks like this:
![image](https://user-images.githubusercontent.com/190571/65808276-1b188f00-e148-11e9-9d9d-60abd35eec38.png)
This PR
![image](https://user-images.githubusercontent.com/190571/65808286-29ff4180-e148-11e9-9789-2febe1cf1a4d.png)
Master
![image](https://user-images.githubusercontent.com/190571/65808386-ad209780-e148-11e9-8e98-e16c12467264.png)

Reply:
Yep, that is what I am going for - the aim is to phase out the non-unified targets in releases going forward (i.e. only Unified Targets and 'exotic' non-unified targets should be included in later 4.1.x releases).
For now, having access to the legacy targets provides a fallback for users if there are problems with Unified Targets, but in a future release we'll probably want to merge them into the 'normal' target selection, and give users non-unified targets for releases before 4.1.0-RC1, and Unified Targets after.
At this point we should also look into switching to using releases for the actual Unified Target .hex for every target - right now I am just not sure how best to do this. Speculatively loading all 140 (for now) Unified Target configurations seems to be a non-ideal way. Maybe we can generate a file containing a list of all Unified Target configurations with their required Unified Target .hex as part of the CI process in the `unified-targets` repository?

Reply:
Generating a file sounds good, if we do it as part of the CI process, where does it go? Oh we have options for that. My vote is github pages https://docs.travis-ci.com/user/deployment/pages/

Reply:
I was thinking of a future CI process that would push any changes to master to something that is more suited as a content delivery network as well.
But even without this, or having CI modify the repository in any form, adding a script that re-generates the index file, requiring target maintainers to update the index file (by script or manually) as part of every pull request, and using CI to fail pull requests that fail to do so should work just fine as a first incarnation.